From b6efd29f61cc65d119459129bef01fa9c597f0ab Mon Sep 17 00:00:00 2001
From: Shane Freeder <theboyetronic@gmail.com>
Date: Tue, 22 Mar 2022 14:56:44 +0000
Subject: [PATCH] Don't send exceptions to the client during kicks, etc

The info here is often not filtered down and can often contain
internal information during these kicks which server owners are
generally not happy with sharing, i.e. potential IP addresses, etc

We'll have to dump the cause to the console manually to ensure that
data is still sent, but, a wider fix here is probably needed which
allows for retaining much of the overall context here, i.e. who
was this exception assocated with?

diff --git a/proxy/src/main/java/net/md_5/bungee/UserConnection.java b/proxy/src/main/java/net/md_5/bungee/UserConnection.java
index 76ab395f..b7d363b8 100644
--- a/proxy/src/main/java/net/md_5/bungee/UserConnection.java
+++ b/proxy/src/main/java/net/md_5/bungee/UserConnection.java
@@ -399,7 +399,8 @@ public final class UserConnection implements ProxiedPlayer
 
     private String connectionFailMessage(Throwable cause)
     {
-        return groups.contains( "admin" ) ? Util.exception( cause, false ) : cause.getClass().getName();
+        bungee.getLogger().log(Level.WARNING, "Error occurred processing connection for " + this.name + " " + Util.exception( cause, false )); // Waterfall
+        return ""; // Waterfall
     }
 
     @Override
-- 
2.35.3

